sync/sync.go

```diff
@@ -322,13 +331,44 @@ func (s *Syncer[H]) requestHeaders(
 
 // storeHeaders updates store with new headers and updates current syncStore's Head.
 func (s *Syncer[H]) storeHeaders(ctx context.Context, headers ...H) error {
```

Comment:
Why is this check done in `storeHeaders`? I would have thought this would happen during verification (that we check if it's the height of the subjective head and compare the hashed)